test/unit/OSTest.cc

```diff
@@ -12,8 +12,8 @@ class OSTest : public testing::Test {
       : os(config::SimInfo::getConfig()["CPU-Info"]["Special-File-Dir-Path"]
                .as<std::string>()),
         proc_elf(simeng::kernel::LinuxProcess(cmdLine)),
-        proc_hex(simeng::span<char>(reinterpret_cast<char*>(demoHex),
-                                    sizeof(demoHex))) {}
+        proc_hex(simeng::span(reinterpret_cast<const uint8_t*>(demoHex),
```

Comment:
Add TODO for future: Don't type-pun when we can just rewrite demoHex in the correct type.

Reply:
Sure, but keep in mind that only punning to `uint_8` is legal here, casting to any other type is UB.

Reply:
Resolving with no action taken